pkg/utils/tests/tests.go

```diff
@@ -68,3 +68,19 @@ func RequireSignal(t *testing.T, ch <-chan struct{}, failMsg string) {
 		t.Fatal(failMsg)
 	}
 }
+
+// SkipShort skips tb during -short runs, and notes why.
+func SkipShort(tb testing.TB, why string) {
+	if testing.Short() {
+		tb.Skipf("skipping: %s", why)
+	}
+}
+
+// SkipShortDB skips tb during -short runs, and notes the DB dependency.
+func SkipShortDB(tb testing.TB) {
```

Comment:
I'm not sure if we should introduce this pattern here. It was added because of the core repo's default dependency on an external DB to run tests, which we do not want to encourage by default.

Reply:
Once the evm extraction is done most of the database tests we currently have will be in a separate evm repo that can't import `chainlink`.   At a minimum, `chainlink`, `chainlink-evm`, and `chainlink-solana` will all need to import this function from somewhere... but every relay will need it eventually, assuming we need to run CCIP on it.  In fact, chainlnk-cosmos also needs it--presently all of the unit tests for tx mgr in the cosmos repo are manually commented out which is not a good situation. IMO we should fix that so they call a function like this.

Reply:
This is the first I've heard that we might want to get rid of external db dependencies in tests.  We can get by in most tests by mocking the ORM, but I don't see how we could test the ORM's without having an external db with all of the right tables set up. Has anyone proposed a plan to move away from that?

Reply:
> At a minimum, chainlink, chainlink-evm, and chainlink-solana will all need to import this function from somewhere

EVM seems exceptional since it has historical baggage coming from core, but this is trivial to copy in that case. I don't see why we would introduce this problem to solana or anything else though.

I am not proposing that we do not test against external DBs. The point is just that a developer should be able to run `go test` without any extra baggage. If there are dependencies required, then typically you opt-in to those tests by e.g. including a build tag, setting a flag/env-var, etc.

Reply:
Given that we have the in-memory version, I don't think we need this short-DB association. However, since the in-memory version might not work for every test, we could use a similar `SkipExternalDB` sort of mechanism to skip those tests when the memory DB is used.

Reply:
> I am not proposing that we do not test against external DBs. The point is just that a developer should be able to run go test without any extra baggage.

Ok, I thought you were saying something different.  I think this might explain some confusion I had in the other thread I responded to here--I hadn't refreshed my browser in a while so hadn't read this.

I just always keep the same db set up locally, so for the most part have never had any issues with it.  The only problem I've run into occasionally is forgetting to run `make testdb` when I upgrade to a new version of chainlink. Usually it becomes obviously fairly quickly tho when tests start failing.

I'm not opposed to the long term goal of trying to allow devs to run tests without having a db set up. But I have a feeling if we get there it would have to be pretty far in the future. I'm not sure an in-memory db would ever be able to substitute for all of the types of tests we need. It's possible we could do that for some of the relays in the future, but trying to do that for Solana would not be possible within the time frame of when this needs to be released. I suppose we could keep it as-is for now and update it after we've got the db isolation stuff done, but I don't think we have much of an idea of what duckdb's capabilities are--that would require some research to see if it's even feasible.

Reply:
My main point:  unless and until we get to a point where we know that 100% of the tests can be run in-memory, there will still be a need for `SkipShortDB()`

Reply:
Alright, making it as flexible as possible sounds good to me.

I think that criterion suggests that we should add both `SkipShortDB()` and the helper function you recommended to `chainlink-common`.  Let each modules decide which one if either they need.  Not exposing either of them is being opinionated.

Reply:
My point is that this library should not be misleading anyone in to thinking that `-short` has anything at all to do with databases. That is a per project decision, and is used as a workaround for bad UX that we have been stuck with. This repo should not promote it.